### PR TITLE
Added TCP LAN WAN Examples #4279

### PR DESCRIPTION
**Pull request by MiguelBarro**

### CODE REVIEW

File: docs/tcp_LAN_WAN_transport.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+TCP Transport
+==============
+
+Fast RTPS transport layer could be adapted to specific network requirements. UDP transport is optimal for localhost applications but in order to interact within a LAN or WAN a TCP transport may be required. Here we explain how Shapes Demo should be set up in order to fit some network specific layouts.
```
Comment (richiware) [inline docs/tcp_LAN_WAN_transport.rst]: UDP is optimal for localhost? Not only for that, also for local networks. TCP must be used when jumps between networks (with routers in the middle) are needed. Please, try to write this idea.

File: docs/tcp_LAN_WAN_transport.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+LAN configuration
+-----------------
+
+This is a common scenario where several computers share the same LAN network. In this case one of the computers should set up Shapes Demo as a server and all the others as clients. Let's suppose that the server computer LAN IP address is 192.168.2.49 then all clients instances of Shapes Demo in the other computers must specify this IP Server address. In this case we select as TCP port 5100 but any other available TCP port is valid. The server computer firewall must allow inbound traffic on the selected port.
```
Comment (richiware) [inline docs/tcp_LAN_WAN_transport.rst]: The common scenario is not this. Usually in LAN is used UDP, but we offer the example of TCP in a LAN network.

File: docs/tcp_LAN_WAN_transport.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+.. image:: LAN_settings_options.png
+   :scale: 100 %
+   :alt: LAN settings options
+   :align: center
+  
+   
+WAN configuration
+-----------------
+
+In this scenario the server computer does not share network with its clients but its reachable through a WAN IP address. This may happen if the server and clients are in a different LANs of the same WAN like WWW. In order to test this scenario we used the following network layout:
```
Comment (richiware) [inline docs/tcp_LAN_WAN_transport.rst]: This is the common scenario for TCP.

File: docs/firststept.rst [hunk 4]
```diff
@@ -67,11 +67,34 @@ Other Options
...
 .. image:: options.png
-   :scale: 100 %
+   :scale: 75 %
    :alt: Options Window
    :align: center
+   
+   
+The user can customize several aspects of ShapesDemo operation:
+
+- **Transport Protocol:** UDP is the default transport protocol for Fast RTPS but TCP functionality is available. In order to use TCP we must be aware of its point-to-point connection nature: one of the ShapesDemo instance must be a TCP server and all the others must be TCP clients. To use TCP follow the next steps:
```
[the remark this answers is not in the document]
Comment (MiguelBarro) [reply docs/firststept.rst]: I'm updating from my local repo. This point is update already.

File: docs/firststept.rst [hunk 5]
```diff
@@ -67,11 +67,34 @@ Other Options
...
+	
+	+ Push the *Stop* button in order to end UDP use. This will automatically remove all publishers and subscribers from this instance.
+	
+	+ To create a TCP LAN server push the corresponding button and fill the *Server Port* editbox with an available port.
```
[the remark this answers is not in the document]
Comment (MiguelBarro) [reply docs/firststept.rst]: Too much Windows GUI programming ;-) I'm updating from my local repo. This point is update already.

File: docs/firststept.rst [hunk 6]
```diff
@@ -67,11 +67,34 @@ Other Options
...
+	
+	+ To create a TCP WAN server push the corresponding button:
+	
+		- fill the *WAN IP* editbox with the WAN router address.
```
[the remark this answers is not in the document]
Comment (MiguelBarro) [reply docs/firststept.rst]: Too much Windows GUI programming ;-) I'm updating from my local repo. This point is update already.

File: docs/firststept.rst [hunk 7]
```diff
@@ -67,11 +67,34 @@ Other Options
...
+		- fill the *Server Port* editbox with an available TCP port.
```
[the remark this answers is not in the document]
Comment (MiguelBarro) [reply docs/firststept.rst]: Too much Windows GUI programming ;-) I'm updating from my local repo. This point is update already.

File: docs/firststept.rst [hunk 8]
```diff
@@ -67,11 +67,34 @@ Other Options
...
+		
+		Note that the router NAT and computer firewall settings must allow external connections to the server port.
+	
+	+ To create a TCP client push push the corresponding button:
+	
+		- fill the *Server IP* editbox with the IP address of the server.
+		- if client and server do not share the same net because the server is behind a NAT the WAN IP address of the server gateway must be specified.
+		- fill the *Server port* editbox with the corresponding server listening port.
```
[the remark this answers is not in the document]
Comment (MiguelBarro) [reply docs/firststept.rst]: I'm updating from my local repo. This point is update already.

File: docs/time_based_filter.rst [hunk 9]
```diff
@@ -1,5 +1,5 @@
 Time Based Filter
-================
+===================
```
[the remark this answers is not in the document]
Comment (MiguelBarro) [reply docs/time_based_filter.rst]: I'm updating from my local repo. This point is update already.

File: docs/tcp_LAN_WAN_transport.rst (hunk 1 above)
Comment (MiguelBarro) [reply docs/tcp_LAN_WAN_transport.rst]: I try to rephrase it as:
UDP transport is optimal for local network applications but in order to interact on a Wide Area Network (WAN), where one or several NAT mappings may be enforced, a TCP transport would be more suitable.

File: docs/tcp_LAN_WAN_transport.rst (hunk 3 above)
Comment (MiguelBarro) [reply docs/tcp_LAN_WAN_transport.rst]: I try to rephrase it as:
This is the typical TCP scenario where the server computer does not share network with its clients but its reachable through a WAN IP address.

**Pull request merged**